File: C++/1805. Number of Different Integers in a String.cpp
class Solution {
public:
    int numDifferentIntegers(string w) {
        int a,m;
        string s="";
        set<string>x;
        for(char c:w){
            if(c>='0'&&c<='9')
            s+=c;
            else{
                if(s.size()!=0){
                    m=0;
                    while(s[m]=='0'&& m < s.size())
                    m++;                
                x.insert(s.substr(m));}
                s="";
            }
        }
        if(s.size()!=0){
                    m=0;
                    while(s[m]=='0'&& m < s.size())
                    m++;
               x.insert(s.substr(m));}
        return x.size();
    }
};

File: JAVA/1805. Number of Different Integers in a String.java
class Solution {
    public int numDifferentIntegers(String w) {
        String s="";
        int m;
        Set<String>x=new HashSet<>();
        for(int i=0;i<w.length();i++){
            if(w.charAt(i)>='0'&&w.charAt(i)<='9')
            s+=w.charAt(i);
            else{
                if(s.length()!=0){
                    m=0;
                    while(m<s.length()&&s.charAt(m)=='0')
                    m++;
                    x.add(s.substring(m));
                }
                s="";
            }
        }
        if(s.length()!=0){
                    m=0;
                    while(m<s.length()&&s.charAt(m)=='0')
                    m++;
                    x.add(s.substring(m));
                }
                return x.size();
    }
}